iron-list.html

```diff
@@ -1096,11 +1101,9 @@
       if (isIndexRendered) {
         inst = this.modelForElement(this._physicalItems[this._getPhysicalIndex(idx)]);
       } else if (offscreenItem) {
-        // This is only needed for hybrid mode.
-        inst = this.modelForElement(offscreenItem)[this.indexAs] === idx ?
-            this.modelForElement(offscreenItem) : null;
+        inst = this.modelForElement(offscreenItem);
       }
-      if (!inst || (this._collection && inst.__key__ !== key)) {
+      if (!inst || (!IS_V2 && inst.__key__ !== key) || (IS_V2 && inst[this.indexAs] !== idx)) {
```

Comment:
I made some changes to simplify the check. so yes, it can just be `inst[this.indexAs] !== idx`.